homeassistant/components/mqtt/__init__.py

```diff
@@ -1247,6 +1265,18 @@ def device_info(self):
         return device_info_from_config(self._device_config)
 
 
+@websocket_api.websocket_command(
+    {vol.Required("type"): "mqtt/device/remove", vol.Required("device_id"): str}
+)
+@websocket_api.async_response
+async def websocket_remove_device(hass, connection, msg):
```

Comment:
We should also remove the retained discovery config from MQTT.

Reply:
This is done:
 - Any retained MQTT discovery topics for entities are cleared from the broker
  https://github.com/home-assistant/home-assistant/blob/e170a11547b716de8cfe46a99473f4dbfcb7f62b/homeassistant/components/mqtt/__init__.py#L1199-L1204
- Any retained MQTT discovery topics for triggers are cleared from the broker
  https://github.com/home-assistant/home-assistant/blob/e170a11547b716de8cfe46a99473f4dbfcb7f62b/homeassistant/components/mqtt/device_trigger.py#L233-L241